Description: Fix build for Android using clang compiler
Author: Boris Pek <tehnick-8@yandex.ru>
Last-Update: 2017-12-15

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 28f8651..a42c8e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -399,7 +403,9 @@ ELSE()
         # '-O0 -gstabs+' helps the currently buggy GDB port
         # Too late to manipulate ENV: SET(ENV{CFLAGS} "$ENV{CFLAGS} -mandroid")
         # Not using _INIT variables, they seem to be used internally only
-        SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mandroid")
+        IF ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+            SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mandroid")
+        ENDIF()
         SET(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -O0 -gstabs+")
     ENDIF()
 ENDIF()
